### PR TITLE
feat(crypto-messages): add schemas

### PR DESCRIPTION
**Pull request by oXtxNt9U**
## Summary

Add schemas for proposals, prevotes and precommits

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

**Comment by oXtxNt9U** [comment]
They might still change and also are not getting applied yet in the factory.

File: packages/crypto-consensus-bls12-381/source/schemas.ts [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
 
 export const schemas = {
-	publicKey: {
+	consensusPublicKey: {
 		...originalSchemas.publicKey,
 		$id: "consensusPublicKey",
 	},
+	consensusSignature: {
+		$id: "consensusSignature",
+		allOf: [{ $ref: "alphanumeric" }, { maxLength: 192, minLength: 192 }],
```
Comment (sebastijankuzner) [inline packages/crypto-consensus-bls12-381/source/schemas.ts:10]: Can be merged with original schema.

File: packages/crypto-messages/source/schemas.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export const schemas: Record<
+	"proposal" | "prevote" | "precommit" | "proposalLockProof" | "validatorBitmap",
+	AnySchemaObject
+> = {
+	precommit: {
+		$id: "precommit",
+		properties: {
+			blockId: { $ref: "blockId" },
+			height: { minimum: 1, type: "integer" },
+			round: { minimum: 1, type: "integer" },
```
Comment (sebastijankuzner) [inline packages/crypto-messages/source/schemas.ts:13]: Round minimum is 0.

File: packages/crypto-messages/source/schemas.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		},
+		required: ["type", "height", "round", "validatorIndex"],
+		type: "object",
+	},
+	prevote: {
+		$id: "prevote",
+		properties: {
+			blockId: { $ref: "blockId" },
+			height: { minimum: 1, type: "integer" },
+			round: { minimum: 1, type: "integer" },
```
Comment (sebastijankuzner) [inline packages/crypto-messages/source/schemas.ts:26]: Round minimum is 0.

File: packages/crypto-messages/source/schemas.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		},
+		required: ["type", "height", "round", "validatorIndex"],
+		type: "object",
+	},
+	proposal: {
+		$id: "proposal",
+		properties: {
+			block: {
+				properties: {
+					serialized: { $ref: "alphanumeric" },
```
Comment (sebastijankuzner) [inline packages/crypto-messages/source/schemas.ts:39]: Can hex be used?

File: packages/crypto-messages/source/schemas.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+				},
+				required: ["serialized"],
+				type: "object",
+			},
+			height: { minimum: 1, type: "integer" },
+			round: { minimum: 1, type: "integer" },
```
Comment (sebastijankuzner) [inline packages/crypto-messages/source/schemas.ts:45]: Round minimum is 0.

File: packages/crypto-messages/source/schemas.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+			},
+		},
+		required: ["signature", "validators"],
+		type: "object",
+	},
+	validatorBitmap: {
+		$id: "validatorBitmap",
+		items: { type: "boolean" },
+		maxItems: 51,
+		minItems: 0, // TODO: milestone
```
Comment (sebastijankuzner) [inline packages/crypto-messages/source/schemas.ts:68]: Min should match max. Set 51 for now.

File: packages/crypto-messages/source/schemas.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+		$id: "precommit",
+		properties: {
+			blockId: { $ref: "blockId" },
+			height: { minimum: 1, type: "integer" },
+			round: { minimum: 1, type: "integer" },
+			signature: { $ref: "consensusSignature" },
+			type: { enum: [Contracts.Crypto.MessageType.Precommit] },
+			validatorIndex: { maximum: 50, minimum: 0, /* TODO: milestone */ type: "integer" },
+		},
+		required: ["type", "height", "round", "validatorIndex"],
```
Comment (sebastijankuzner) [inline packages/crypto-messages/source/schemas.ts:18]: Signature can be required for all schemas.

File: packages/crypto-consensus-bls12-381/source/schemas.ts (hunk 1 above)
Comment (oXtxNt9U) [reply packages/crypto-consensus-bls12-381/source/schemas.ts:10]: Signatures don't seem to have any schemas

**Pull request merged**